shared/health/src/test/java/io/pravega/shared/health/HealthServiceTests.java

```diff
@@ -0,0 +1,232 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.health;
+
+import io.pravega.test.common.AssertExtensions;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.rules.Timeout;
+
+import io.pravega.shared.health.TestHealthIndicators.SampleHealthyIndicator;
+
+/**
+ * The {@link HealthServiceTests} encapsulates much of the same processes that {@link HealthEndpoint} performs, so
+ * an explicit test class for the former is skipped.
+ */
+@Slf4j
+public class HealthServiceTests {
+
+    @Rule
+    public final Timeout timeout = new Timeout(60, TimeUnit.SECONDS);
+
+    HealthService service;
+
+    HealthServiceFactory factory;
+
+    private void start() {
+        factory = new HealthServiceFactory();
+        service = factory.createHealthService(true);
+    }
+
+    private void stop() {
+        service.clear();
```

Comment:
This class has a `close` method defined on it. Call it. 

Fix your other tests too.